Sources/Private/CoreAnimation/Layers/ShapeLayer.swift

```diff
@@ -354,15 +350,17 @@ extension Collection {
 
 /// A group of `ShapeItem`s that should be rendered together as a single unit
 struct ShapeRenderGroup {
-  /// The items in this group that render `CGPath`s
+  /// The items in this group that render `CGPath`s.
+  /// Valid shape render groups must have at least one path-drawing item.
   var pathItems: [ShapeItemLayer.Item] = []
   /// Shape items that modify the appearance of the shapes rendered by this group
   var otherItems: [ShapeItemLayer.Item] = []
 }
 
 extension Array where Element == ShapeItemLayer.Item {
-  /// Splits this list of `ShapeItem`s into groups that should be rendered together as individual units
-  var shapeRenderGroups: [ShapeRenderGroup] {
+  /// Splits this list of `ShapeItem`s into groups that should be rendered together as individual units,
+  /// plus the remaining items that were not included in any group.
+  var shapeRenderGroups: (groups: [ShapeRenderGroup], unusedItems: [ShapeItemLayer.Item]) {
```

Comment:
Maybe something like this to make it more clear than `groups.groups`?
```suggestion
  var shapeRenderGroups: (used: [ShapeRenderGroup], unused: [ShapeItemLayer.Item]) {
```

Reply:
Good idea